packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -60,6 +60,11 @@ export interface RemoveProps {
     blockPropagate?: boolean;
 }
 
+export interface TagManagerProps {
+    initialTags?: Tags;
+    autoScalingGroup?: boolean;
```

Comment:
This property feels awfully specific. What is its purpose, exactly? Is it because Tag rendering is slightly different for ASGs (With the `propagateAtLaunch` setting)?

If so, what if the next type of resource comes along that has a slightly different tag rendering? Do we add a boolean for every different type of tag rendering?

Can we make this mechanism more generally applicable? A potential easy solution is to make a subclass of TagManager that's specific to AutoScalingGroups that renders tags slightly different. We can even keep it in the `aws-autoscaling` package.

Reply:
To be precies, seems that `resolve()` seems to end with the pattern:

```ts
return Object.keys(tags).map( key => some_function(key) );
```

How about making that `some_function` a call to a member function on `TagManager`?

Reply:
Ok -- yes I'll create that method and make it protected.